docs/src/content/docs/utilities/add-virtual-import.mdx

```diff
@@ -0,0 +1,58 @@
+---
+title: addVirtualImport
+description: Allows adding Vite Virtual Modules easily to access data across your integration.
+---
+
+`addVirtualImport` allows you to create modules and store state across your integration. For example:
+
+```ts title="integration/index.ts"
+import type { AstroIntegration } from "astro";
+import { addVirtualImport } from "astro-integration-kit";
+
+export default function myIntegration(): AstroIntegration {
+    return {
+        name: "my-integration",
+        hooks: {
+            "astro:config:setup": ({ updateConfig }) => {
+                addVirtualImport({
+                    name: 'virtual:my-integration/config', // this will now be importable with `virtual:my-integration/config`
+                    content: `export default ${ JSON.stringify({foo: 'bar'}) }`,
+                    updateConfig,
+                })
+            }
+        }
+    }
+}
+```
+
+So to read this config anywhere else in your integration is easy:
+
+```ts title="my-integration/src/pages/index.astro"
+---
+import config from 'virtual:my-integration/config',
+
+console.log(config.foo) // 'bar'
+---
+```
+
+Now you can easily share state across your integration 🎉!
+
+:::caution
+Virtual Modules defined with `addVirtualImport` cannot be referenced inside the `astro-integration-kit` integration utils.
```

Comment:
```suggestion
Virtual modules defined with `addVirtualImport` cannot be referenced inside your integration code, for example inside `integration/index.ts`. That's because integration code is running before vite injects the virtual imports
```